client/src/components/admin/SubmissionsPage/SurveySubmissionsChart.tsx

```diff
@@ -0,0 +1,224 @@
+}
+
+type Data = {
+  id: number;
+  options: {
+    id: number;
+    text: string;
+    count: number;
+  }[];
+};
+
+type Range = {
+  min: number;
+  max: number;
+};
+
+function buildNumericRange(range: Range, answersList: AnswerEntry[]): number[] {
+  const maxBuckets = 10;
+  const minBuckets = 5;
+
+  if (answersList[0].type === 'slider') {
+    return Array.from({ length: 11 }, (_, i) => i);
+  }
+
+  // If range min/max are provided, use those. Otherwise find min/max values
+  const min = Math.floor(
+    range.min ??
+      (answersList.reduce(
+        (min, answer) => ((answer.value as number) < min ? answer.value : min),
+        null as number,
+      ) as number),
+  );
+  const max = Math.ceil(
+    range.max ??
+      (answersList.reduce(
+        (max, answer) => ((answer.value as number) > max ? answer.value : max),
+        null as number,
+      ) as number),
+  );
+
+  // For short ranges, use step of 1
+  if (max - min <= maxBuckets && max - min > minBuckets) {
+    return Array.from({ length: max - min + 1 }, (_, i) => min + i);
+  }
+
+  // increases / decreases bucket count until all items fit in bucketCount * allowedBuckets
+  const calcBucketSize = (r: number, bucketSize: number) => {
+    let bucketCount = Math.ceil(r / bucketSize);
+    while (bucketCount < minBuckets) {
+      bucketSize /= 2;
+      bucketCount = Math.ceil(r / bucketSize);
+    }
+    while (bucketCount > maxBuckets) {
+      bucketSize *= 2;
+      bucketCount = Math.ceil(r / bucketSize);
+    }
+    return [bucketCount, bucketSize];
+  };
+
+  const [numOfBuckets, step] = calcBucketSize(
+    max - min,
+    // Starting position is first power of 10 to fit full range
+    Math.pow(10, Math.floor(Math.log10((max - min) / maxBuckets))),
+  );
+  return Array.from({ length: numOfBuckets }, (_, i) => min + step * i);
+}
+
+const CustomizedAxisTick: FunctionComponent<any> = (props: any) => {
+  const { x, y, payload } = props;
+  const tickValue =
+    payload.value.length > 12
+      ? payload.value.slice(0, 11) + '...'
+      : payload.value;
+  return (
+    <g transform={`translate(${x},${y})`}>
+      <text
+        x={0}
+        y={0}
+        dy={16}
+        textAnchor="end"
+        fill="#666"
+        transform="rotate(-35)"
+      >
+        {tickValue}
+      </text>
+    </g>
+  );
+};
+
+export default function Chart({ submissions, selectedQuestion }: Props) {
+  const [chartWidth, setChartWidth] = useState(240);
+  const { surveyLanguage } = useTranslations();
+  const theme = useTheme();
+
+  const answerData = useMemo(() => {
+    if (!selectedQuestion) return;
+
+    const questionAnswers = submissions
+      .reduce(
+        (answers, submission) => [...answers, ...submission.answerEntries],
+        [] as AnswerEntry[],
+      )
+      .filter((answer) => answer.sectionId === selectedQuestion.id);
+    if (!questionAnswers) return;
+
+    let base: Data;
+    switch (selectedQuestion?.type) {
```

Comment:
[nit] `?.` shouldn't be required here, because the function has returned on the first line if it's falsy.